strax/plugin.py

```diff
@@ -847,3 +847,50 @@ def cleanup(self, iters, wait_for):
             for s in savers:
                 s.close(wait_for=wait_for)
         super().cleanup(iters, wait_for)
+
+
+@export
+class CutPlugin(Plugin):
+    """Generate a plugin that provides a boolean for a given cut specified by 'cut_by'"""
+    save_when = SaveWhen.NEVER
+
+    def __init__(self):
+        super().__init__()
+
+        _name = strax.camel_to_snake(self.__class__.__name__)
+        if not hasattr(self, 'provides'):
+            self.provides = _name
+        if not hasattr(self, 'cut_name'):
+            self.cut_name = _name
+        if not hasattr(self, 'cut_description'):
+            _description = _name
+            if 'cut_' not in _description:
+                _description = 'Cut by ' + _description
+            else:
+                _description = " ".join(_description.split("_"))
+            self.cut_description = _description
+
+    def infer_dtype(self):
+        dtype = [(self.cut_name, np.bool, self.cut_description)]
```

Comment:
Does not it have to be np.bool_ ? https://numpy.org/doc/stable/user/basics.types.html

Reply:
thanks, didn't know that, will change it!